x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/PolicyStepsRegistry.java

```diff
@@ -59,6 +59,8 @@ public PolicyStepsRegistry() {
     @SuppressWarnings({ "unchecked", "rawtypes" })
     public void update(ClusterState currentState, Client client, LongSupplier nowSupplier) {
         IndexLifecycleMetadata meta = currentState.metaData().custom(IndexLifecycleMetadata.TYPE);
+        assert meta != null : "IndexLifecycleMetadata cannot be null when updating the policy steps registry";
```

Comment:
I've seen this pattern of assertion elsewhere in the codebase. I am not sure how I feel about it, but I chose to try it out here to be self-documenting. Currently this method is only ever called once, and it is behind a null check